### PR TITLE
Sebawoz02/ci cmake

### PR DESCRIPTION
**Pull request by sebawoz02**
1. Added a simple CMakeLists.txt file to make it easier to build and manage the project.

2. Included temporary files and directories that we need to get the project structure started. These will help us organize our codebase as we move forward. We can revisit and clean these up as the project matures.

3. Added a GitHub Actions workflow file (`main_ci.yml`) to automate continuous integration. This workflow will help us ensure our code quality and build stability as we develop the project.

### CODE REVIEW

File: src/tmp.cpp
Comment (kociap) [inline src/tmp.cpp:1]: what is this for?

File: scripts/config_env.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+fi
+
+if ! dpkg -l | grep -q libglfw3-dev; then
+    echo "GLFW development libraries are not installed. Installing..."
+    sudo apt update
+    sudo apt-get install -y libglfw3 libglfw3-dev
+else
+    echo "GLFW development libraries are already installed."
+fi
+
```
Comment (kociap) [inline scripts/config_env.sh:19]: double newline

File: include/project/tmp.hpp
Comment (kociap) [inline include/project/tmp.hpp:1]: what is this for?

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+set(CMAKE_CXX_COMPILER "g++")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic -lGL")
+
+find_package(OpenGL REQUIRED)
+find_package(glfw3 REQUIRED)
+
+include_directories(${OPENGL_INCLUDE_DIR} include)
+
+file(GLOB SOURCES "src/*.cpp")
```
Comment (kociap) [inline CMakeLists.txt]: list files explicitly instead of globbing

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.22)
+project(NEBULA)
+
+set(CMAKE_CXX_STANDARD 11)
+
+set(CMAKE_CXX_COMPILER "g++")
```
Comment (kociap) [inline CMakeLists.txt]: what if I use clang?

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.22)
+project(NEBULA)
```
Comment (kociap) [inline CMakeLists.txt]: why scream?

File: CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+set(CMAKE_CXX_STANDARD 11)
```
Comment (kociap) [inline CMakeLists.txt]: 20

File: app/main.cpp [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    glfwSwapBuffers(window);
+    glfwPollEvents();
+  }
+
+  glfwTerminate();
+
+  return 0;
+}
+
+
```
Comment (kociap) [inline app/main.cpp]: multiple newlines

**Review by kociap — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 3 above)
Comment (sebawoz02) [reply CMakeLists.txt]: USE_CLANG option added. ON - clang, OFF - g++

File: include/project/tmp.hpp
Comment (sebawoz02) [reply include/project/tmp.hpp:1]: Github does not allow you to save empty folders, so I put temporary files in src/ and include/ to indicate that the code to be implemented later should be in these folders. I'm removing them for now, but the first person writing the code should remember to organize the project in this way.

File: CMakeLists.txt (hunk 3 above)
Comment (kociap) [reply CMakeLists.txt]: CMAKE_CXX_COMPILER may be set at setup time via cli defines. no need for custom options. the flags we will use are compatible with both gcc and clang.

File: CMakeLists.txt [hunk 7]
```diff
@@ -1,17 +1,28 @@
...
+option(USE_CLANG "Use clang instead of g++" OFF)
+if(USE_CLANG)
+    message(STATUS "Using clang as compiler")
+    set(CMAKE_CXX_COMPILER "clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++20 -Weverything -Wno-c++98-compat")
+else()
+    # Use g++ by default
+    message(STATUS "Using g++ as compiler")
+    set(CMAKE_CXX_COMPILER "g++")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++20 -Wall -Wextra -pedantic")
```
Comment (kociap) [inline CMakeLists.txt]: -std=c++20 will be set by cmake. setting it manually prevents the project from being configured by cmake

**Review by kociap — CHANGES_REQUESTED** [review]

**Pull request closed without merging**